packages/analytics-client-common/src/global-scope.ts

```diff
@@ -2,6 +2,12 @@
 /* Only file allowed to access to globalThis, window, self */
 
 export const getGlobalScope = (): typeof globalThis | undefined => {
+  // This should only be used for integrations with Amplitude that are not running in a browser environment
+  //   We need to specify the name of the global variable as a string to prevent it from being minified
+  const ampIntegrationContextName = 'ampIntegrationContext' as keyof typeof globalThis;
+  if (typeof globalThis[ampIntegrationContextName] !== 'undefined') {
```

Comment:
should you also check to make sure `globalThis` is also not undefined?

Reply:
Can we also add some tests for this? I was surprised that we didn't have tests for this file at atll but the CI passes which requires 100% coverage.

Reply:
Hmm maybe the commons folder is excluded? Will write tests on Monday!

Reply:
It's not included in test coverage because it has no corresponding test file. I'm ok with no tests as I didn't find an easy way to mock `globalThis` to be undefined. 

Here is what I tried `global-scope.test.ts`:
```
import { getGlobalScope } from '../src/global-scope'

describe('getGlobalScope', () => {
  test('should return globalThis if defined', () => {
    expect(globalThis).toBeDefined();
    expect(getGlobalScope()).toBe(globalThis);
  });

  test('should return window if globalThis is undefined and window is defined', () => {
    const originalGlobalThis: typeof globalThis = globalThis;
    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
    // @ts-ignore
    globalThis = undefined;

    expect(globalThis).toBeUndefined();
    expect(window).toBeDefined();
    expect(getGlobalScope()).toBe(window);

    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
    // @ts-ignore
    globalThis = originalGlobalThis;
  });
});
```

The error suggests that something in the Jest setup or the mocking process is interfering with Jest's internal state or its matchers.
```
Test suite failed to run
TypeError: Cannot read properties of undefined (reading 'Symbol($$jest-matchers-object)')
```